src/seedu/addressbook/Main.java

```diff
@@ -84,10 +85,13 @@ private void runCommandLoopUntilExitCommand() {
         do {
             String userCommandText = ui.getUserCommand();
             command = new Parser().parseCommand(userCommandText);
-            CommandResult result = executeCommand(command);
-            recordResult(result);
-            ui.showResultToUser(result);
-
+            try {
+                CommandResult result = executeCommand(command);
+                recordResult(result);
+                ui.showResultToUser(result);
+            } catch (StorageOperationException e) {
+                start(new String[0]);//restart the program
```

Comment:
This is one of the ways to handle the exception. Think what would make the user experience better in terms of not losing the data entered.